batch/batch/front_end/front_end.py

```diff
@@ -225,8 +225,15 @@ async def get_healthcheck(_) -> web.Response:
 
 
 @routes.get('/api/v1alpha/version')
-async def rest_get_version(_) -> web.Response:
-    return web.Response(text=version())
+async def rest_get_version(request: web.Request) -> web.Response:
+    headers = None
+    server_version = version()
+    client_version = request.headers.get("X-Hail-Version")
+    if client_version is not None and client_version != server_version:
+        headers = {
+            "X-Hail-Deprecated": f"The version of the Batch Client ({client_version}) does not match the version of Batch running on the server ({server_version}). Please upgrade your Hail version to the latest release."
+        }
+    return web.Response(text=server_version, headers=headers)
```

Comment:
I think warning whenever there is a newer release is probably too noisy. For now lets just warn when they are specifically depending on deprecated functionality.

If we did do this, we'd also need to be sure we're only comparing the release version, not the commit hash. Otherwise, since we deploy batch on every commit, every user would be out of date as soon as any PR merges after a release.

Reply:
Good point! I had honestly not realized we deploy Batch on every commit. If at some point down the line we come up with a policy for e.g. how many patch versions the difference between client and server can be, we can always reintroduce the version checking, but for now I agree it doesn't need to be there. Removed!